### ISSUE TITLE
Implement import tracking

### ISSUE THREAD

**Issue by FozzieHi** [PyCQA/flake8-bugbear#357]
Following reports like #356, I think these errors are prevalent across multiple lints and one way to solve them would be to use some form of import tracking.

I've come up with some examples below, but this is just a very rough draft.

`from itertools import groupby`
```python
imported_modules = {
    "itertools": {
        "name": "itertools",
        "imports": {
            "groupby": "groupby"
        }
    }
}
```

`from itertools import groupby as groupby_as`
```python
imported_modules = {
    "itertools": {
        "name": "itertools",
        "imports": {
            "groupby": "groupby_as"
        }
    }
}
```

`import itertools`
```python
imported_modules = {
    "itertools": {
        "name": "itertools"
    }
}
```

`import itertools as itertools_as`
```python
imported_modules = {
    "itertools": {
        "name": "itertools"
    }
}
```

Then in the lint to get the `itertools` id you'd use `imported_modules["itertools"]["name"]`, and to get the `groupby` id you'd use `imported_modules["itertools"]["imports"]["groupby"]`.

Is this something that we would be interested in adding?

**Comment by FozzieHi** [comment]
@cooperlees What do you think about this?

**Comment by cooperlees** [comment]
So, just so I understand, you just want to make a generic API to add import tracking to multiple checks in a central place? Any other main goals I missed here?

**Comment by FozzieHi** [comment]
Yup, just want to effectively create a method to get the import alias for a certain module, which we can then use in the checks.

**Comment by cooperlees** [comment]
Sounds worth adding to me, especially if we're already using it in multiple places (I forget).
